src/cloud-cli/meltano/cloud/cli/run.py

```diff
@@ -40,17 +40,25 @@ async def run_project(
 @click.argument("job_or_schedule")
 @click.option(
     "--deployment",
-    required=True,
     help="The name of the Meltano Cloud deployment to run in.",
 )
 @pass_context
 @run_async
 async def run(
     context: MeltanoCloudCLIContext,
     job_or_schedule: str,
-    deployment: str,
+    deployment: str | None = None,
 ) -> None:
     """Run a Meltano project in Meltano Cloud."""
+    if deployment is None:
+        if context.config.default_deployment_name is not None:
+            deployment = context.config.default_deployment_name
+        else:
+            raise click.UsageError(
+                "A deployment name is required. Please specify "
+                "one with the '--deployment' option, or specify a default"
+                "deployment by running 'meltano cloud deployment use'.",
+            )
```

Comment:
Wdyt of using an if-expression here instead of nested if statements?

```suggestion
    deployment = (
        deployment
        if deployment is not None
        else context.config.default_deployment_name
    )
    if deployment is None:
        raise click.UsageError(
            "A deployment name is required. Please specify "
            "one with the '--deployment' option, or specify a default"
            "deployment by running 'meltano cloud deployment use'.",
        )
```

Reply:
That's much nicer. Thank you for the suggestion.